app/forms/support_interface/application_forms/edit_reference_feedback_form.rb

```diff
@@ -2,30 +2,24 @@ module SupportInterface
   module ApplicationForms
     class EditReferenceFeedbackForm
       include ActiveModel::Model
-      include ValidationUtils
 
-      attr_accessor :feedback
-      attr_accessor :audit_comment
-
-      attr_reader :reference
-      attr_reader :application_form
+      attr_accessor :feedback, :audit_comment
 
       validates :feedback, presence: true, word_count: { maximum: 500 }
       validates :audit_comment, presence: true
 
-      def initialize(application_form, reference)
-        @application_form = application_form
-        @reference = reference
-
-        super(feedback: @reference.feedback)
+      def self.build_from_reference(reference)
+        new(feedback: reference.feedback)
       end
 
-      def save!
-        @reference.update!(
+      def save(reference)
+        return false unless valid?
+
+        reference.update!(
           feedback: feedback,
           audit_comment: audit_comment,
         )
-        SubmitReference.new(reference: @reference).save!
+        SubmitReference.new(reference: reference).save!
```

Comment:
Do you think this should be called in the controller? It's not clear that this behaviour is going on from just calling save on the reference.

Reply:
Thinking about it. I don't think we should be calling this at all really. 

![image](https://user-images.githubusercontent.com/42515961/102225199-d6108200-3ede-11eb-8518-956855b22265.png)

It does 3 things:

1. Update the reference to feedback_completed which is great
2. Emails the candidate to say received
3. Emails the referee

This would mean that if all we want to do is update the feedback, the emails still get sent.

I think we should just call r`eference.feedback_provided!`

Reply:
`  def initialize(reference:, edited_on_support: false)
    @reference = reference
  end

  def save!
    reference_feedback_provided!
    cancel_feedback_requested_references if enough_references_have_been_provided?
    CandidateMailer.reference_received(reference).deliver_later if edited_on_support
    RefereeMailer.reference_confirmation_email(application_form, reference).deliver_later if edited_on_support
  end
`

Reply:
> Do you think this should be called in the controller? It's not clear that this behaviour is going on from just calling save on the reference.

I agree, it's worth pulling this out to make it clearer what is going on.

Reply:
> Open to better suggestions though.

I've pushed a working implementation similar to your suggestion. Let me know what you think